Assignment6Test.java

```diff
@@ -0,0 +1,249 @@
+package assignment6;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static assignment6.Assignment6.rearrange;
+import static assignment6.Assignment6.rearrangeGreedy;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Assignment6Test {
+    private int[] given;
+    private int[] desired;
+    /**
+     * Executes a plan (list of movements) on an arrangements of cars.
+     *
+     * @param given current map of the parking lot
+     * @param plan list of movements to be executed on the parking lot
+     */
+    public static void executeMoves(int[] given, List<Move> plan) {
+        plan.forEach((m) -> {
+            given[m.getTo()] = m.getCar();
+            given[m.getFrom()] = 0;
+        });
+    }
+
+    @Before
+    public void setUp() {

```

Comment:
Why is there a setup here? It seems like we always use a different `given` and `desired` so we can get rid of this setup